src/corecel/CMakeLists.txt

```diff
@@ -65,6 +65,14 @@ if(CELERITAS_USE_CUDA)
   )
 endif()
 
+if(CELERITAS_USE_HIP)
+  list(APPEND SOURCES
+    sys/ScopedProfiling.hip.cc
+  )
+  find_library(ROCTX_LIBRARY roctx64 REQUIRED HINTS ${ROCM_PATH}/lib64)
```

Comment:
This might be better as part of the `celeritas_device_toolkit` (see CMakeLists in the paernt directory). Also, and I'm assuming the answer is no, but is there a significant reason *not* to link this in when building HIP? I.e., does it inject any symbols that change the behavior of the HIP API functions, or could it be missing from some HIP installations? (In which case we need extra compile-time logic for the profile functions...)

Reply:
Do you mean linking hip against roctx64? I don't think there's an option to do that. It's definitely possible that this library is missing from some installation as it has to be installed separately (at least from my experience installing hip from spack and from [here](https://rocm.docs.amd.com/en/latest/understand/cmake_packages.html#rocm-cmake-packages)) 

We probably should make it optional and have the ScopedProfiling print a warning if the library was not found?

Reply:
Yeah. OK if it has to be installed separately, we'll need to have some configuration options that describe the system setup. Let's do one of these two:
-  create a `celeritas_sys_config.h.in` using `celeritas_config.h.in` as a reference, and define `CELERITAS_HAVE_ROCTX` there (in the future we may have to add more special conditions for optional libraries)
- add a private `target_compile_definitions` with `-DCELERITAS_HAVE_ROCTX=1`? since we probably only need these configuration options inside corecel.

Reply:
Should be good now, sorry for all the CI triggers 😅